### ISSUE TITLE
Interval expression not be evaled in date_arith

### ISSUE THREAD

**Issue by zyguan** [pingcap/tidb#1140]
https://github.com/pingcap/tidb/blob/master/evaluator/builtin_time.go#L465

`nodeInterval.Interval` should be evaled before calling `GetDatum()`.

---

Please answer these questions before submitting your issue. Thanks!
1. What version of Go are you using (`go version`)?
   go1.6.1
2. What operating system and processor architecture are you using (`go env`)?
   linux_amd64
3. What did you do?
   `SELECT DATE_ADD('2010-10-10', INTERVAL 1+1 DAY);`
4. What did you expect to see?

```
+------------------------------------------+
| DATE_ADD('2010-10-10', INTERVAL 1+1 DAY) |
+------------------------------------------+
| 2010-10-12                               |
+------------------------------------------+
1 row in set (0.00 sec)

```
1. What did you see instead?

```
+------------------------------------------+
| DATE_ADD('2010-10-10', INTERVAL 1+1 DAY) |
+------------------------------------------+
| NULL                                     |
+------------------------------------------+
1 row in set (0.00 sec)

```

**Comment by shenli** [comment]
@zyguan Thanks for your report!

**Comment by zyguan** [comment]
solved by #2368
